zebrad/src/config.rs

```diff
@@ -4,8 +4,25 @@
 //! application's configuration file and/or command-line options
 //! for specifying it.
 
+use std::path::PathBuf;
+
 use serde::{Deserialize, Serialize};
 
+/// Returns true if a leaf key name should be considered sensitive and blocked
+/// from environment variable overrides.
+fn is_sensitive_leaf_key(leaf_key: &str) -> bool {
+    let lower = leaf_key.to_ascii_lowercase();
+    // Centralized, case-insensitive suffix-based deny-list
+    lower.ends_with("password")
+        || lower.ends_with("secret")
+        || lower.ends_with("token")
+        // Block raw cookies only if a field is literally named "cookie".
+        // (Paths like cookie_dir are not affected.)
+        || lower.ends_with("cookie")
+        // Only raw private keys; paths like *_private_key_path are not affected.
+        || lower.ends_with("private_key")
+}
```

Comment:
It would be nice to have these in a constant so they can be referenced below.

```suggestion
/// Centralized, case-insensitive suffix-based deny-list to ban setting config fields with
/// environment variables if those config field names end with any of these suffixes.
const DENY_CONFIG_KEY_SUFFIX_LIST: [&str; 5] = [
    "password",
    "secret",
    "token",
    // Block raw cookies only if a field is literally named "cookie".
    // (Paths like cookie_dir are not affected.)
    "cookie",
    // Only raw private keys; paths like *_private_key_path are not affected.
    "private_key",
];

/// Returns true if a leaf key name should be considered sensitive and blocked
/// from environment variable overrides.
fn is_sensitive_leaf_key(leaf_key: &str) -> bool {
    let key = leaf_key.to_ascii_lowercase();
    DENY_CONFIG_KEY_SUFFIX_LIST.iter().any(|deny_suffix| key.ends_with(deny_suffix))
}
```